client/Dockerfile

```
@@ -0,0 +1,54 @@
+# syntax=docker.io/docker/dockerfile:1
+
+# REFERRANCE: NEXT JS Docker documentation
+
+FROM node:18-alpine AS base
+
+# Install dependencies only when needed
+FROM base AS deps
+RUN apk add --no-cache libc6-compat
+WORKDIR /app
+
+# Install dependencies based on the preferred package manager
+COPY package.json yarn.lock* package-lock.json* pnpm-lock.yaml* .npmrc* ./
+RUN \
+  if [ -f yarn.lock ]; then yarn --frozen-lockfile; \
+  elif [ -f package-lock.json ]; then npm ci; \
```

when in production we can omit the dev using
```suggestion
  elif [ -f package-lock.json ]; then npm ci --omit=dev; \
```

---

Got it bhaiya.